openvdb/openvdb/points/PointRasterizeFrustum.h

```diff
@@ -0,0 +1,1805 @@
+                        }
+                    } else {
+                        if (mComputeMax && !forceSum) {
+                            valueAccessor.modifyValue(ijk, MaxOpT(newValue));
+                        } else {
+                            valueAccessor.modifyValue(ijk, SumOpT(newValue));
+                        }
+                        if (weightAccessor) {
+                            weightAccessor->modifyValue(ijk, SumOpT(weightValue));
+                        }
+                    }
+                }
+            }
+        };
+
+        auto modifyVoxelOp = [&](const Coord& ijk, const double scale, const AttributeT& attributeScale)
+        {
+            doModifyVoxelOp(ijk, scale, attributeScale, /*isTemp=*/false, /*forceSum=*/false);
+        };
+
+        auto sumVoxelOp = [&](const Coord& ijk, const double scale, const AttributeT& attributeScale)
+        {
+            doModifyVoxelOp(ijk, scale, attributeScale, /*isTemp=*/false, /*forceSum=*/true);
+        };
+
+        auto doModifyVoxelByDistanceOp = [&](const Coord& ijk, const double scale, const AttributeT& attributeScale,
+            const double distanceSqr, const double radiusSqr, const bool isTemp)
+        {
+            if (distanceSqr >= radiusSqr)   return;
+            if (isBool) {
+                valueAccessor.modifyValue(ijk, TrueOp(scale));
+            } else {
+                double distance = std::sqrt(distanceSqr);
+                double radius = std::sqrt(radiusSqr);
+                double result = 1.0 - distance/radius;
+                doModifyVoxelOp(ijk, result * scale, attributeScale, isTemp, /*forceSum=*/false);
+            }
+        };
+
+        auto modifyVoxelByDistanceOp = [&](const Coord& ijk, const double scale, const AttributeT& attributeScale,
+            const double distanceSqr, const double radiusSqr)
+        {
+            doModifyVoxelByDistanceOp(ijk, scale, attributeScale, distanceSqr, radiusSqr, /*isTemp=*/false);
+        };
+
+        auto modifyTempVoxelByDistanceOp = [&](const Coord& ijk, const double scale, const AttributeT& attributeScale,
+            const double distanceSqr, const double radiusSqr)
+        {
+            doModifyVoxelByDistanceOp(ijk, scale, attributeScale, distanceSqr, radiusSqr, /*isTemp=*/true);
+        };
+
+        typename points::AttributeHandle<AttributeT>::Ptr attributeHandle;
+        using ElementT = typename ValueTraits<AttributeT>::ElementType;
+        typename points::AttributeHandle<ElementT>::Ptr stridedAttributeHandle;
+
+        if (mAttributeIndex != points::AttributeSet::INVALID_POS) {
+            const auto& attributeArray = leaf.constAttributeArray(mAttributeIndex);
+            if (attributeArray.stride() == 3) {
+                stridedAttributeHandle = points::AttributeHandle<ElementT>::create(attributeArray);
+            } else {
+                attributeHandle = points::AttributeHandle<AttributeT>::create(attributeArray);
+            }
```

Comment:
What attribute is this using? Only supports a stride of 3? Do the attribute handles not handle this internally?